yamscripts.yml

```diff
@@ -10,54 +10,46 @@ scripts:
+  mochitest-debug: (cd $npm_package_config_mc_dir && ./mach mochitest --jsdebugger browser/extensions/activity-stream/test/functional/mochitest)
 
-  cleanmc: rimraf $npm_package_config_mc_dir/browser/extensions/activity-stream
+# bundle: Build all assets for activity stream
+  bundle:
+    locales: pontoon-to-json --src locales --dest data
+    webpack: webpack --config webpack.system-addon.config.js --display-optimization-bailout
+    css: node-sass --source-map true --source-map-contents content-src/styles -o css
+    html: rimraf prerendered && webpack --config webpack.prerender.config.js && node ./bin/render-activity-stream-html.js
 
 # buildmc: Export the bootstraped add-on to mozilla central
   buildmc:
-    pre: =>cleanmc
-    locales: pontoon-to-json --src locales --dest system-addon/data
-    webpack: webpack --config webpack.system-addon.config.js --display-optimization-bailout
-    css: node-sass --source-map true --source-map-contents system-addon/content-src/styles -o system-addon/css
-    html: rimraf system-addon/prerendered && webpack --config webpack.prerender.config.js && node ./bin/render-activity-stream-html.js
-    # Copy over all of the system-addon directory
-    copy: cpx "system-addon/**/{,.}*" $npm_package_config_mc_dir/browser/extensions/activity-stream
+    pre: rimraf $npm_package_config_mc_dir/browser/extensions/activity-stream/
+    bundle: => bundle
+    copy: rsync --exclude-from .mcignore -a . $npm_package_config_mc_dir/browser/extensions/activity-stream/
     version: node ./bin/update-version.js $npm_package_config_mc_dir/browser/extensions/activity-stream
     stringsExport: cp locales/en-US/strings.properties $npm_package_config_mc_dir/browser/locales/en-US/chrome/browser/activity-stream/newtab.properties
-    copyPingCentre: cpx "system-addon/ping-centre/PingCentre.jsm" $npm_package_config_mc_dir/browser/modules
-    rmextra: rimraf "$npm_package_config_mc_dir/browser/extensions/activity-stream/{{,test/unit/}{content-src,ping-centre},data/locales.json}"
-
-# packagemc: Build stand-alone system add-on xpi
-  packagemc:
-    pre: =>cleanmc
-    build: =>buildmc:webpack && =>buildmc:css && =>buildmc:locales
-    copy: cpx "system-addon/**" dist/system-addon
-    version: node ./bin/update-version.js dist/system-addon
-    process: node ./bin/process-system-addon-for-package.js dist/system-addon
-    xpi: (cd dist/system-addon && zip -r ../activity-stream-system-addon.xpi * )
+    copyPingCentre: cpx "ping-centre/PingCentre.jsm" $npm_package_config_mc_dir/browser/modules
 
 # startmc: Start developing the bootstrapped add-on
   startmc:
     _parallel: true
-    pre: =>prebuildmc
-    copy: =>buildmc:copy -- -w
+    pre: =>buildmc
+    # This copies only the system addon sub-folder; changing anything outside of it will need a full rebuild.
+    copy: cpx "{{,.}*,!(dist|logs|node_modules|ping-centre)/**/{,.}*}" $npm_package_config_mc_dir/browser/extensions/activity-stream/ -w
```

Comment:
If `startmc` is primarily to streamline development process, I think it's okay if we're a bit excessive in copying. So ignoring just `node_modules` will probably be good enough to keep this fast. Otherwise we might try too hard to keep this ignore list exact.